### PR TITLE
Use ReduxForm Fields for domain inputs in Basic Settings.

### PR DESCRIPTION
**Pull request by jseppi**
This fixes the bug with losing focus after the first character is typed in the domain inputs on the Basic Settings page (#1307).

I did so by turning those inputs into custom ReduxForm Fields (code example at https://redux-form.com/7.2.0/examples/fieldlevelvalidation/), similar to the existing `GitHubRepoUrlField` component. Since the new `HttpsUrlField` and the existing `GitHubRepoUrlField` are very similar, they now share a common `RenderUrlField` component.

Validation errors now look like this (instead of being HTML5 validation error tooltips):

screen shot 2017-12-13 at 12 01 15 pm

### CODE REVIEW

File: frontend/components/Fields/HttpsUrlField.jsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { Field } from 'redux-form';
+import { isURL } from 'validator';
+
+import RenderUrlField from './RenderUrlField';
+
+
+export const isHttpsUrlWithoutPath = (value) => {
+  if (value && value.length) {
+    if (!isURL(value, { protocols: ['https'] }) || value.lastIndexOf('/') !== 7) {
```
Comment (el-mapache) [inline frontend/components/Fields/HttpsUrlField.jsx]: Does this method report a valid URL if the two trailing `/` after the `:` are missing?

Comment (jseppi) [reply frontend/components/Fields/HttpsUrlField.jsx]: Nope, it reports that is invalid. Just added a couple additional test expectations to show that in 69/commits/4811c

Comment (el-mapache) [reply frontend/components/Fields/HttpsUrlField.jsx]: if that's the case, do we need the `||` condition here?

Comment (jseppi) [reply frontend/components/Fields/HttpsUrlField.jsx]: that's to make sure that there is no additional path in the url. `7` is the position of the last `/` in `https://`. Any additional would indicate that there is a sub-path, which we don't want.

ie, `https://example.gov` is valid, but `https://example.gov/` and `https://example.gov/whatever` are not valid.

Comment (el-mapache) [reply frontend/components/Fields/HttpsUrlField.jsx]: I didnt see the withoutPath part of the method name, makes sense.

**Pull request merged**